src/Approximations/decompositions.jl

```diff
@@ -63,89 +40,78 @@ projections.
 
 ### Algorithm
 
-For each block a specific `project` method is called, dispatched on the
-`set_type` argument.
+For each block a specific `project` method is called, dispatching on the
+corresponding overapproximation option.
 
 ### Notes
 
-If `directions` is different from `nothing`, the template directions are used
-together with `blocks`.
-Otherwise, if `block_types` is given, the options `set_type` and `blocks` are
-ignored.
+The argument `partition` requires some discussion.
+Typically, the list of blocks should form a partition of the set
+``\\{1, \\dots, n\\}`` represented as a list of consecutive blocks, where ``n``
+is the ambient dimension of set `S`.
 
-### Examples
+However, technically there is no problem if the blocks are not consecutive,
+blocks are missing, blocks occur more than once, or blocks are overlapping.
+This function will, however, stick to the order of blocks, so the resulting set
+must be interpreted with care in such cases.
+One use case is the need of a projection consisting of several blocks.
 
-The `decompose` function supports different options, such as: supplying different
-dimensions for the decomposition, defining the target set of the decomposition,
-or specifying the degree of accuracy of the target decomposition. You can also
-choose to make the approximations in low dimensions using template directions.
-These options are exemplified below.
+For convenience, the argument `block2oa` can also be given as a single `oa`
+option, which is then interpreted as the option for all blocks.
 
-#### Different dimensions
+### Examples
 
-By default, `decompose` returns a Cartesian product of 2D `Hyperrectangle` sets.
-For example:
+This function supports different options: one can specify the target set, the
+degree of accuracy, and template directions.
+These options are exemplified below, where we use the following example.
 
 ```jldoctest decompose_examples
-julia> import LazySets.Approximations:decompose
+julia> using LazySets.Approximations: decompose
 
-julia> S = Ball2(zeros(4), 1.);
+julia> S = Ball2(zeros(4), 1.);  # set to be decomposed (4D unit ball in the 2-norm)
 
-julia> array(decompose(S))
-2-element Array{LazySet{Float64},1}:
- Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
- Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
-```
+julia> P2d = [1:2, [3, 4]];  # a partition with two blocks of size two
```

Comment:
why not either `[[1,2], [3, 4]]` or `[1:2, 3:4]` ?

Reply:
Ha, this was on purpose to show the most popular ways, but I see now that is confusing.